core/src/main/kotlin/org/evomaster/core/search/gene/GeneUtils.kt

```diff
@@ -485,19 +494,34 @@ object GeneUtils {
                     assert(selectedGene.gene !is CycleObjectGene)
                     repairBooleanSelection(selectedGene.gene)
                 }
-            } else {
-                (selectedGene as BooleanGene).value = true
-            }
+            } else
+                if (selectedGene is TupleGene) {
+                    val lastElement = selectedGene.elements.last()
+                    isLastElementInTupleOptional(lastElement)
+                } else
+                    (selectedGene as BooleanGene).value = true
         }
     }
 
-    fun shouldApplyBooleanSelection(gene: Gene) =
-            (gene is OptionalGene && gene.gene is ObjectGene)
-                    || gene is ObjectGene
-                    || (gene is ArrayGene<*> && gene.template is ObjectGene)
-                    || (gene is ArrayGene<*> && gene.template is OptionalGene && gene.template.gene is ObjectGene)
-                    || (gene is OptionalGene && gene.gene is ArrayGene<*> && gene.gene.template is OptionalGene && gene.gene.template.gene is ObjectGene)
-                    || (gene is OptionalGene && gene.gene is ArrayGene<*> && gene.gene.template is ObjectGene)
+    private fun isLastElementInTupleOptional(lastElement: Gene) {
+        if (lastElement is OptionalGene) {
+            lastElement.isActive = true
```

Comment:
this is not valid.
you have a method whose name starts with `is`, which implies check for boolean, which this method does not return.
furthermore, this method has a side effect on the input argument `lastElement`.
name of function must be changed

Reply:
done